packages/utils/src/internals/sitemap.ts

```diff
@@ -0,0 +1,103 @@
+import { createGunzip } from 'node:zlib';
+
+import log from '@apify/log';
+import type { SAXStream } from 'sax';
+import sax from 'sax';
+
+class ParsingState {
+    sitemapUrls: string[] = [];
+    urls: string[] = [];
+    visitedSitemapUrls: string[] = [];
+    context: 'sitemapindex' | 'urlset' | undefined = undefined;
+    loc: boolean = false;
+
+    resetContext() {
+        this.context = undefined;
+        this.loc = false;
+    }
+}
+
+export class Sitemap {
```

Comment:
should have jsdoc too, on classes I'd like to see a basic example of the expected usage